scripts/gulpfiles/appengine_tasks.js

```diff
@@ -84,6 +78,50 @@ function deployAndClean(done) {
       rimraf.sync(demoTmpDir);
     }
   }
+}
+
+/**
+ * Constructs a demo version name based on the version specified in
+ * package.json.
+ */
+function getDemosVersion() {
+  const minorVersion = packageJson.version.split('.')[1];
+  const patchVersion = packageJson.version.split('.')[2];
+  let demoVersion = minorVersion;
+  if (patchVersion != 0) {
+    demoVersion += '-' + patchVersion
+  }
+  return demoVersion;
+}
+
+/**
+ * Deploys files from tmp directory to appengine to version based on the version
+ * specified in package.json and then cleans the tmp directory.
+ */
+function deployAndClean(done) {
+  const demoVersion = getDemosVersion();
+  deployToAndClean(demoVersion);
+  done();
+}
+
+/**
+ * Constructs a beta demo version name based on the current date.
+ */
+function getDemosBetaVersion() {
+  var date = new Date();
```

Comment:
My only concern with this is that it makes it impossible to do multiple deployments on the same day (unless having multiple versions is not a problem? I'm not sure how it works)

So having a command line argument to specify the version number, or a prefix, or something, so that we can specify a unique one if needed, would be helpful. I'm not sure how often we do this or if we plan to stick with the App Engine approach anyway so feel free to ignore this or wait until we decide

Reply:
You can do multiple deployments on the same day, it will overwrite the existing version that is deployed.
Given that we use beta for testing, I think overwriting it is the desired behavior.
I am a fan of this automatically generating the name, because it makes it much more convenient than typing in the version name, but open to discussion. I wont merge this in until after release.

Reply:
Cool, overwriting makes sense then.

I was thinking it would create by default, unless an option was passed in. But it sounds like we don't actually need it.